### DISCUSSION TITLE
Does this addon send every site to Google? If not, how does it automatically recognize language?

### DISCUSSION THREAD

**Discussion by secretmango** [FilipePS/Traduzir-paginas-web#537] [Q&A] [answered]
I would really appreaciate a completely passive mode, that only translates when **I** press the button. 

Otherwise, with all that background recognizing stuff, I am really unsure if not every single site I visit gets sent to Googles servers, making this addon a pretty good tracking tool, even though I am sure this is not intended.

**Comment by FilipePS** [comment, marked as answer]
If the option `Show translation when hovering over *` is checked when hovering over texts, these texts will be translated and a popup will be shown with the translation result.
Whenever you open a website and the extension icon in the address bar turns blue, the website is being translated.

Whenever a translation occurs, the text is sent to google or another translation service (unless the text has been translated before and is cached).

By default no site is translated unless you explicitly translate it. However you can activate sites or language to always translate when opening the site.

So if something translation starts without you interacting with the extension, you either purposely or didn't configure the extension for that.

The simplest way to reset all settings is to uninstall the extension and install it again.

**Reply by secretmango** [reply]
Thank you! So the language recognition is done by the chosen translation service, but only when chosen?

**Reply by FilipePS** [reply]
Language detection to determine if the page should be translated is done offline by the extension using the `i18n.detectLanguage` or `tabs.detectLanguage` API.
